IR with DPR, training only, first cut

franzmpub opened the pull request.

franzmpub commented:
Test coverage is at 47% now, working on resolving it.

franzmpub commented:
Test coverage is at 60%.

avisil reviewed (approved):
LGTM @franzmpub . Please update the main README page as it shows DPR (coming soon) etc. Otherwise, this is ready to get merged else I see any issues reported by anyone else.

franzmpub commented:
> 

@bhavani105 My suggestion is to remove one in https://github.com/primeqa/primeqa/blob/main/primeqa/ir/util/corpus_reader.py, which (to my understanding) is a leftover from previous DPR integration effort.

bhavani105 commented:
> > 
> 
> @bhavani105 My suggestion is to remove one in https://github.com/primeqa/primeqa/blob/main/primeqa/ir/util/corpus_reader.py, which (to my understanding) is a leftover from previous DPR integration effort.

Its being used by BM25 indexing. I think these utils are not specific to DPR

franzmpub commented:
> > > 
> > 
> > 
> > @bhavani105 My suggestion is to remove one in https://github.com/primeqa/primeqa/blob/main/primeqa/ir/util/corpus_reader.py, which (to my understanding) is a leftover from previous DPR integration effort.
> 
> Its being used by BM25 indexing. I think these utils are not specific to DPR

removed dense/dpr_top/util/corpus_reader.py, updated related imports

michaelrglass commented:
There are a number of utility methods not used in this code that could be cut. HypersBase has jsonl_instances and kofn for example. But leaving them doesn't hurt anything.

In DPRSearcher I had some code to deal with older different versions of corpus indexes before standardizing on 'pid' as the id for a passage. With the corpus_reader normalizing these differences all the checks in DPRSearcher for 'pid' vs. 'id' vs. no id are not needed. You could just cut everything other than the code path for 'pid'.

Also in DPRSearcher it looks like the 'include_vectors = False' code path is not implemented and not really needed. I would just cut that unless you have near term plans to make it work.

michaelrglass commented on primeqa/ir/dense/dpr_top/dpr/searcher.py:
```diff
@@ -0,0 +1,262 @@
+                else:
+                    docs = merge_results(query_vectors, self.opts.n_docs_for_provenance)
+
+                if 'pid' in docs[0][0]:
```
You don't need these checks anymore. Passages will always have a pid field.

michaelrglass commented on primeqa/ir/dense/dpr_top/dpr/searcher.py:
```diff
@@ -0,0 +1,262 @@
+                                  'text': [dqk['text'] for dqk in dq]} for dq in docs]
+
+                include_vectors = True # TODO: make this an arg? Note: this is "if not only_docs" in corpus_client.retrieve
+                if include_vectors:
```
I would just cut include_vectors = False as an option

michaelrglass commented on primeqa/ir/dense/dpr_top/torch_util/hypers_base.py:
```diff
@@ -0,0 +1,259 @@
+            except:
+                pass
+
+    def kofn(self, kofn: str):
```
Not used, but doesn't hurt anything.

franzmpub commented:
> Looks good. I commented on a few possible simplifications.

Modified as suggested.

bhavani105 reviewed (approved):
LGTM
